dokka-subprojects/analysis-kotlin-descriptors-ide/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/ide/DokkaResolutionFacade.kt

```diff
@@ -87,6 +87,10 @@ internal class DokkaResolutionFacade(
                     }
                 }
 
+                override fun getProject(): Project? {
+                    throw UnsupportedOperationException()
```

Comment:
Theoretically, we do have some project as a property of DokkaResolutionFacade - I wonder if it should be returned instead? Or if the return type is nullable, maybe `null` could be returned as well

Throwing an exception generally feels kinda scary in analysis, god knows what corner case might cause `getProject()` to be called now or in future versions (assuming kotlin-compiler has access to this object for something)

Reply:
1. on current moment only `get` from `BindingContext` is called on returned object and it's not passed anywhere - so chances of calling this method are minimal. So to fail here we will need to explicitly call `getProject` and so most likely it will fail on first test run.
2. Other methods override there do the same - `throw UnsupportedOperationException()` - and I would say, it's expected behaviour, as they should not be called.

I could change it to `null`, but I don't see a real reason why we should handle this method differently comparing to other ones.

Reply:
> it's not passed anywhere

oh, I thought it was created to then be passed to the compiler or something. If it's only used by us, it's indeed not a problem 👍